scripts/aws.sh

```diff
@@ -0,0 +1,488 @@
+    ParameterKey=AmiId,ParameterValue="${ami_id}" \
+    ParameterKey=EC2Type,ParameterValue="${ec2Type}" \
+    ParameterKey=PublicKeyString,ParameterValue="$(cat "${pub_key}")"
+    
+  #echo "Waiting for stack to be created"
+  aws --region "${region}" cloudformation wait stack-create-complete --stack-name "${stack_name}" 2> "${err_log}" &
+  pid=$!
+  spin "${pid}" "Waiting for stack to be created" "Stack created successfully"
+
+  # shellcheck disable=SC2016
+  instance_id="$(aws --region "${region}" cloudformation describe-stacks --stack-name "${stack_name}" \
+    --query 'Stacks[].Outputs[?OutputKey == `InstanceId`].OutputValue' --output text)"
+
+  aws --region "${region}" ec2 wait instance-status-ok --instance-id "${instance_id}" 2> "${err_log}" &
+  pid=$!
+  spin "${pid}" "Waiting for stack status to be OK" "Stack status OK"
+
+  # shellcheck disable=SC2016
+  public_ip=$(aws --region "${region}" cloudformation describe-stacks --stack-name "${stack_name}" \
+    --query 'Stacks[].Outputs[?OutputKey == `PublicIp`].OutputValue' --output text)
+
+  rm -f "${cf_tpl_file}"
+  
+  echo "PUBLIC IP: ${public_ip}"
+}
+
+action_delete() {
+  aws --region "${region}" cloudformation delete-stack --stack-name "${stack_name}"
+	aws --region "${region}" cloudformation wait stack-delete-complete --stack-name "${stack_name}" 2> "${err_log}" &
+  pid=$!
+  spin "${pid}" "Waiting for stack to delete" "Stack deleted"
+	exit 0
+}
+
+action_describe() {
+  aws --region "${region}" cloudformation describe-stack-events --stack-name "${stack_name}" --output json | less
+	exit 0
+}
+
+action_logs() {
+  # shellcheck disable=SC2016
+  instance_id="$(aws --region "${region}" cloudformation describe-stacks --stack-name "${stack_name}" \
+    --query 'Stacks[].Outputs[?OutputKey == `InstanceId`].OutputValue' --output text)"
+  
+  aws ec2 get-console-output --instance-id "${instance_id}" --output text
+}
+
+usage() {
+  cat - <<EOF
+Script for AWS stack management
+
+Usage:
+
+aws.sh (create|delete|describe|logs) \ 
+            --stack-name <name> \ 
+            [--inst-type <type>] \ 
+            [--region <region>] \ 
+            [--pub-key <path>]       
+
+Create
+
+  aws.sh create --stack-name <name> --inst-type <type> [--region <region>] [--pub-key <path>]
+
+Delete|Describe|Logs
+
+  aws.sh (delete|describe|logs) --stack-name <name> [--region <region>]
+
+Arguments:
+  --stack-name <name>:  The name of the stack.
+  [--inst-type <type>]: (create only) The type of instance to create (e.g. t3.large, c5n.metal).
+  [--region <region>]:  (create only) The region where the stack should be created. Defaults to eu-west-1.
+  [--pub-key <path>]:   (create only) The path to a .pub file. Defaults to /home/${USER}/.ssh/id_rsa.pub.
+
+EOF
+}
+
+action="$1"
+shift
+
+case "${action}" in
+  create|delete|describe|logs)
+    ;;
+  *)
+    usage
+    exit 1
+    ;;
+esac
+
+stack_name=""
+inst_type=""
+region=eu-west-1
+pub_key="/home/${USER}/.ssh/id_rsa.pub"
```

Comment:
We should not assume user home directory is rooted at `/home`
```suggestion
pub_key="~/.ssh/id_rsa.pub"
```

Reply:
There's a high chance `~` won't be interpreted when passed to awscli. We can use `${HOME}` instead